src/core.js

```diff
@@ -245,47 +197,63 @@ define([
+        var _this = this;
         this.data.core.QUESTIONS_IN_TOOLBAR = [];
-        this.data.core.QUESTION_TYPE_TO_GROUP = {};
-
-        _.each(this._getQuestionGroups(), function (groupData) {
-            var groupSlug = groupData.group[0];
 
+        _.each(_this.getQuestionGroups(), function (groupData) {
             var getQuestionData = function (questionType) {
                 var mugType = _this.data.core.mugTypes[questionType],
                     questionData = [
-                        questionType, 
-                        mugType.typeName, 
+                        questionType,
+                        mugType.typeName,
                         mugType.icon
                     ];
 
                 _this.data.core.QUESTIONS_IN_TOOLBAR.push(questionType);
-                _this.data.core.QUESTION_TYPE_TO_GROUP[questionType] = groupSlug;
                 return questionData;
             };
 
             groupData.questions = _.map(groupData.questions, getQuestionData);
-            if (groupData.related && groupData.related.length) {
-                groupData.related = _.map(groupData.related, getQuestionData);
-            }
+        });
 
-            groupData.group[2] = groupData.group[2] || 
-                _this.data.core.mugTypes[groupData.group[0]].icon;
-            $questionGroupContainer.append(
-                new QuestionTypeGroup(groupData, _this));
+        _this.$f.find(".fd-add-question").after($(add_question({
+            groups: _.map(_this.getQuestionGroups(), function(groupData) {
+                var defaultMug = _this.data.core.mugTypes[groupData.group[0]];
+                return {
+                    name: groupData.group[1] || defaultMug.typeName,
+                    defaultQuestion: {
+                        slug: groupData.group[0],
+                        name: groupData.group[1] || defaultMug.typeName,
+                        icon: groupData.group[2] || defaultMug.icon,
+                    },
+                    questions: _.map(groupData.questions, function(questionType) {
+                        var mugType = _this.data.core.mugTypes[questionType];
+                        return {
+                            slug: questionType,
+                            name: mugType.typeName,
+                            icon: mugType.icon,
+                        };
+                    }),
+                };
+            }),
+        })));
+
+        _this.$f.on('click', '.fd-question-type', function (e) {
+            if (!$(this).hasClass('disabled')) {
+                _this.addQuestion($(this).data('qtype'));
+            }
+            e.preventDefault();
         });
+    };
 
+    fn._init_toolbar = function () {
         var $saveButtonContainer = this.$f.find('.fd-save-button');
         this.data.core.saveButton.ui.appendTo($saveButtonContainer);
     };
 
-    fn._getQuestionGroups = function () {
+    fn.getQuestionGroups = function () {
```

Comment:
I'm not sure if it matters here, but I think this change has some effect in the plugin system (see [base.js:35](https://github.com/dimagi/Vellum/blob/5960d9d3ed90d11a327ec5488c45667a83599fee/src/base.js#L35))